src/foam/nanos/auth/CachingAuthService.java

```diff
@@ -69,18 +72,27 @@ public CachingAuthService(AuthService delegate, String[] extraDAOsToListenTo) {
 
   @Override
   public boolean check(foam.core.X x, String permission) {
-    if ( x == null || permission == null ) return false;
+    if ( x == null || permission == null ) {
+      StdoutLogger.instance().debug(getClass().getSimpleName() + ".check", "x or permission not provided", permission);
+      return false;
+    }
 
     Permission p = new AuthPermission(permission);
 
     User user = getUserFromContext(x);
     Map<String, Boolean> map = getPermissionMap(user);
 
-    if ( map.containsKey(p.getName()) ) return map.get(p.getName());
+    Logger logger = Loggers.logger(x, this);
+
+    if ( map.containsKey(p.getName()) ) {
+      logger.debug("Found cached permission check result", user, p.getName(), map.get(p.getName()));
```

Comment:
Don't do the map.get() twice.
Are we sure we want to do this? This will be prohibitively expensive.